### PR TITLE
Added FileFromUrl

### PR DESCRIPTION
**Pull request by mvidner**
[FATE#316796](https://fate.suse.com/316796)
https://github.com/yast/yast-registration/pull/241

Note the implementation is ugly, because it was simply copied from the other package, and the test is only basic, aiming only to ensure that the move succeded and that the interface works.

I already have better tests (in https://github.com/yast/yast-autoinstallation/pull/193 ) but I first want to get the API in place so that https://github.com/yast/yast-autoinstallation/pull/195 and https://github.com/yast/yast-registration/pull/241 (which is the ultimate goal) can proceed.

After the tests get improved, I want to improve the implementation too.

### CODE REVIEW

File: test/file_from_url_test.rb [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+    # adaptor for existing tests
+    def Get(scheme, host, urlpath, localfile)
+      get_file_from_url(scheme: scheme, host: host, urlpath: urlpath,
+                        localfile: localfile,
+                        urltok: {}, destdir: "/destdir")
+    end
+  end
+
+  # avoid BuildRequiring a package that we stub entirely anyway
```
Comment (lslezak) [inline test/file_from_url_test.rb]: A nice trick! :+1:

**Comment by schubi2** [comment]
LGTM

File: src/lib/transfer/file_from_url.rb [hunk 2]
```diff
@@ -0,0 +1,578 @@
...
+    #
+    # @param scheme    [String] cifs, nfs, device, usb, http, https, ...
+    # @param host      [String]
+    # @param urlpath   [String]
+    # @param localfile [String] destination filename
+    # @param urltok    [Hash{String => String}] same url as above, but better
+    # @param destdir   [String] chroot (with crazy juggling)
+    #
+    # @return [Boolean] true on success
+    def get_file_from_url(scheme:, host:, urlpath:, localfile:,
```
Comment (lslezak) [inline src/lib/transfer/file_from_url.rb]: NTH: I know this was just a copy & paste but the `get_file_from_url` method is ~500 lines long and that would deserve some refactoring...

**Comment by lslezak** [comment]
LGTM

File: src/lib/transfer/file_from_url.rb (hunk 2 above)
Comment (mvidner) [inline src/lib/transfer/file_from_url.rb]: Yes, yes! I have already started doing it, the right way: by covering it with tests. See  yast/yast-autoinstallation#193

**Pull request merged**